### PR TITLE
add filter support for >, <, >=, <=, !=

### PR DESCRIPTION
**Pull request by jmachen**
Filter targets can now be evaluated using any of the following operators >, <, >=, <=, != by using the desired operator as the seperator between column name and filter value, like so:

`http://localhost:9090/api/v1/cars?&order_direction=asc&order_by=weight&filter=weight>=1000`

This constructs a GORM query like the following:

```(sql)
SELECT * FROM "cars" WHERE "weight" >= '1000' AND "cars"."deleted_at" IS NULL ORDER BY "weight" LIMIT 10
```

A colon or any combination of two operators not explicitly supported, but matching this regex:

```(regex)
[:<>!=]{1,2}
```

...is treated as an = filter operation.

### CODE REVIEW

File: gin-gorm-filter.go [hunk 1]
```diff
@@ -97,13 +97,26 @@ func filterField(field reflect.StructField, phrase string) clause.Expression {
 	} else {
 		paramName = columnName
 	}
-	re, err := regexp.Compile(fmt.Sprintf(`(?m)%v:(\w{1,}).*`, paramName))
+	re, err := regexp.Compile(fmt.Sprintf(`(?m)%v([:<>!=]{1,2})(\w{1,}).*`, paramName))
```
Comment (ActiveChooN) [inline gin-gorm-filter.go:100]: I would suggest using explicit group not to handle unsupported operator combinations.
```suggestion
	re, err := regexp.Compile(fmt.Sprintf(`(?m)%v(:|!=|>|<|>=|<=)(\w{1,}).*`, paramName))
```

**Review by ActiveChooN — COMMENTED** [review]
@jmachen, thanks a lot for your contribution! Would you mind adding some tests for your feature?

**Comment by jmachen** [comment]
Sure thing, sorry I should have waited until I had written them.

File: gin-gorm-filter.go (hunk 1 above)
Comment (jmachen-shiptracks) [reply gin-gorm-filter.go]: Ah, that's fair.  I'll make the change.

Comment (jmachen) [reply gin-gorm-filter.go]: Done!

File: gin-gorm-filter.go [hunk 2]
```diff
@@ -97,13 +97,26 @@ func filterField(field reflect.StructField, phrase string) clause.Expression {
 	} else {
 		paramName = columnName
 	}
-	re, err := regexp.Compile(fmt.Sprintf(`(?m)%v:([^,]*).*`, paramName))
+	re, err := regexp.Compile(fmt.Sprintf(`(?m)%v(:|!=|>|<|>=|<=)(\w{1,}).*`, paramName))
```
Comment (ActiveChooN) [inline gin-gorm-filter.go:100]:
```suggestion
	re, err := regexp.Compile(fmt.Sprintf(`(?m)%v(:|!=|>|<|>=|<=)([^,]*).*`, paramName))
```

**Comment by jmachen** [comment]
Unit test issues should be resolved thanks to the merge corrections.

Local results:
```
go test -v .   
=== RUN   TestRunSuite
=== RUN   TestRunSuite/TestFiltersAndSearch
=== RUN   TestRunSuite/TestFiltersBasic
=== RUN   TestRunSuite/TestFiltersGreaterThan
=== RUN   TestRunSuite/TestFiltersGreaterThanOrEqualTo
=== RUN   TestRunSuite/TestFiltersLessThan
=== RUN   TestRunSuite/TestFiltersLessThanOrEqualTo
=== RUN   TestRunSuite/TestFiltersMultipleColumns
=== RUN   TestRunSuite/TestFiltersNoFilterConfig
=== RUN   TestRunSuite/TestFiltersNotEqualTo
=== RUN   TestRunSuite/TestFiltersNotFilterable
=== RUN   TestRunSuite/TestFiltersOrderBy
=== RUN   TestRunSuite/TestFiltersPaginateOnly
=== RUN   TestRunSuite/TestFiltersSearchable
--- PASS: TestRunSuite (0.01s)
    --- PASS: TestRunSuite/TestFiltersAndSearch (0.00s)
    --- PASS: TestRunSuite/TestFiltersBasic (0.00s)
    --- PASS: TestRunSuite/TestFiltersGreaterThan (0.00s)
    --- PASS: TestRunSuite/TestFiltersGreaterThanOrEqualTo (0.00s)
    --- PASS: TestRunSuite/TestFiltersLessThan (0.00s)
    --- PASS: TestRunSuite/TestFiltersLessThanOrEqualTo (0.00s)
    --- PASS: TestRunSuite/TestFiltersMultipleColumns (0.00s)
    --- PASS: TestRunSuite/TestFiltersNoFilterConfig (0.00s)
    --- PASS: TestRunSuite/TestFiltersNotEqualTo (0.00s)
    --- PASS: TestRunSuite/TestFiltersNotFilterable (0.00s)
    --- PASS: TestRunSuite/TestFiltersOrderBy (0.00s)
    --- PASS: TestRunSuite/TestFiltersPaginateOnly (0.00s)
    --- PASS: TestRunSuite/TestFiltersSearchable (0.00s)
PASS
ok      github.com/ActiveChooN/gin-gorm-filter  0.336s
```

**Review by ActiveChooN — APPROVED** [review]
LGTM, thank you for your contribution!

**Pull request merged**